### DISCUSSION TITLE
Rehype-raw and components

### DISCUSSION THREAD

**Discussion by sergeunity** [remarkjs/remark#1123] [Q&A] [answered]
Hi all. 

I'm trying to add rehype-raw to all people to use <br> in md. 
1st  question, is it a good way ? 

2nd, when I enable the rehype-raw my custom components are ignored, is it not compatible  ?

```jsx
 <ReactMarkdown
      remarkPlugins={[
        remarkTab,
        remarkGfm,
        remarkNotice,
        remarkBreak,
      ]}
      rehypePlugins={[
        rehypeSlug,
        anchorHeading,
        rehypeRaw, 
      ]}
      remarkRehypeOptions={{
        allowDangerousHtml: true,
        handlers: {
         notice: (h, node) => h(node, "notice", node, all(h, node)),
          tabList: (h, node) => h(node, "tabList", node, all(h, node)),
          tab: (h, node) => h(node, "tab", node, all(h, node)),
        },
      }}
      components={components}
    >
   {props.markdown}
    </ReactMarkdown>
```
The result is 
<img width="427" alt="image" src="https://user-images.githubusercontent.com/52750122/223522733-fc1f5442-4b53-4152-bfdd-87efb3ace315.png">

**Comment by wooorm** [comment, marked as answer]
> Is it a good way ?

The readme states that it is not a great idea: https://github.com/remarkjs/react-markdown#appendix-a-html-in-markdown. It defeats the purpose of this project.

> custom components

You do not show enough code. There is no `components` variable in your slice of code. So it’s hard to help you.
But it is likely what you can see in the screenshot: `tablist` is used. Because in HTML, casing does not exist. `rehype-raw` does that. It follows HTML. It’s not `tabList`. It’s `tablist`.

**Reply by sergeunity** [reply]
Thanks a lot. It was just the casing. 

> > Is it a good way ?
> 
> The readme states that it is not a great idea: https://github.com/remarkjs/react-markdown#appendix-a-html-in-markdown. It defeats the purpose of this project.

It is for an internal doc, made by internal dev. We trust them :)
